### PR TITLE
Removing vms_compat dependency from LoadISISNexus

### LINKED ISSUE #27695: Remove vms_compat dependency from LoadISISNexus2
To load the new Muon Nexus files (V2) the LoadISISNexus2 loader will be utilised. In order to use this loader, the dependence of the LoadISISNexus2 on the isis_vms_compat block will need to be addressed.

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**

To load the new Muon Nexus files (V2) the LoadISISNexus2 loader will be utilised. In order to use this loader, the dependence of the LoadISISNexus2 on the isis_vms_compat was removed in this PR. Primarily this PR extracts a number of functions into a set of helper functions.

The following information was removed from the sample logs, as it was deemed either unnecessary, or was no longer being written to the nexus files.

This PR represents the first stage of developing the new Muon Nexus V2 loader #27701.

Variable | Code description|
|-------------------------|-----------|
run_header | Provides a brief description of the run. e.g instrument  and run number |
duration | Actual run duration
duration units | Scalar for run duration
duration_freq | Test interval for duration
dmp | Dump interval
dmp units | Scalar for dump interval
dmp_freq | Interval for dump
dur_wanted | Requested run duration
dur_secs | Actual run duration
mon_sum1 | Monitor sum1
mon_sum2 | Monitor sum2
mon_sum3 | Monitor sum3

**To test:** 
Outside of loading a workspace and checking the correct entries are there, you can
1. On a master branch load a workspace with LoadISISNexus and then save it
```
workspace_1 = LoadISISNexus("LOQ49886.nxs") 
SaveNexus(workspace_1,"LoadISISBefore.nxs")
```
2. Switch to this branch and load the same workspace with LoadISISNexus and compare the workspaces
```
workspace_1 = LoadISISNexus("LOQ49886.nxs") # load workspace with updated alg
workspace_2 = LoadNexus("LoadISISBefore.nxs") # load the workspace created from old alg
workspace_1.equals(workspace_2, 1e-9) # both should be equal
```
3. Additionally, the LoadISISNexus algorithm should now also load the Muon V2 file (you'll have to ask me for an example file)
4. Test that the Muon V2 file is successfully loaded, e.g plot some spectra, check sample logs.

Fixes #27695 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
@stephen, I've got two emu nxs_v2 files will these do for the muon tests?
- EMU00101521.nxs_v2
- EMU00103738.nxs_v2

**Comment by StephenSmith25** [comment]
> 
> 
> @StephenSmith25 , I've got two emu nxs_v2 files will these do for the muon tests?
> 
>     * EMU00101521.nxs_v2
> 
>     * EMU00103738.nxs_v2

Yeah, those should do, I've been testing it with EMU00103638.nxs_v2.

**Comment by NickDraper** [comment]
Ok I ran a full ws comparison and it didn't quite match, we have some missing sample logs
image


Missing:

dmp
dmp_freq
dmp_units
dur
dur_freq
dur_secs
dur_wanted
durunits
mon_sum1
mon_sum2
mon_sum3
run_header

Admittedly I'm not sure if any of these are important.

``` python 
wsRaw = mtd["LOQ49886"]
wsB4 = mtd["LoadISISBefore"]
raw_log_names = []
run = wsRaw.run()
for log in run.getLogData():
    raw_log_names.append(log.name)
    
    
b4_log_names = []
run = wsB4.run()
for log in run.getLogData():
    b4_log_names.append(log.name)

print("Missing:\n")
for name in b4_log_names:
    if name not in raw_log_names:
        print(name)
        
        
print("\nAdded:\n")
for name in raw_log_names:
    if name not in b4_log_names:
        print(name)
```

**Comment by StephenSmith25** [comment]
These variables were removed, as they were either available elsewhere, or deemed unnecessary. I think it discussed a while back (possibly January) and Martyn agreed at that time. These variables were chosen as I couldn’t see anything in Mantid that was using them and most seemed obscure.

**Review by NickDraper — CHANGES_REQUESTED** [review]
Trying to load Muon nxs_v2 files using the Load algorithm fails.  It is selecting the LoadMuonNexus algorithm as the loader.  You'll need to look at the confidence methods of LoadMuonNexus2 and LoadISISNexus, to make sure that LoadISISNexus wins for these files.
```
Error in execution of algorithm LoadMuonNexus:
Histogram: size mismatch of data.

Error in execution of algorithm Load:
Histogram: size mismatch of data.
```

We also need a nxs v2 file added to the LoadLotsOfFiles System test.


Otherwise really good, and I'm glad to see the back of the vms_compat block.

**Comment by StephenSmith25** [comment]
> 
> 
> Trying to load Muon nxs_v2 files using the Load algorithm fails. It is selecting the LoadMuonNexus algorithm as the loader. You'll need to look at the confidence methods of LoadMuonNexus2 and LoadISISNexus, to make sure that LoadISISNexus wins for these files.
> 
> ```
> Error in execution of algorithm LoadMuonNexus:
> Histogram: size mismatch of data.
> 
> Error in execution of algorithm Load:
> Histogram: size mismatch of data.
> ```
> 
> We also need a nxs v2 file added to the LoadLotsOfFiles System test

So currently this PR is the halfway point of loading the Muon v2 files and I've got another PR to put up which introduces a new algorithm to load the muon v2 files, which essentially calls LoadISISNexus (to do the bulk of the loading, which is made possible with these changes) and then does the Muon Specific stuff (e.g grouping, time zero) ontop. So might be best I address those issues in that PR.

**Comment by NickDraper** [comment]
Ok can you add a reference to the issue for that then?

**Review by NickDraper — APPROVED** [review]
So the muon v2 files load into LoadISISNexus directly, is the Load issues and addition of a LoadLotsofFiles nxsV2 file happens in #27701 then this is good to go.

I'm really pleased to see us move on from the VMS days!

**Review by NickDraper — CHANGES_REQUESTED** [review]
One minor thing though as this alters some of our Loading algorithms we do need to mention this in the release notes.

**Comment by StephenSmith25** [comment]
> 
> 
> One minor thing though as this alters some of our Loading algorithms we do need to mention this in the release notes.

That’s true, I had forgot about this, I've pushed a documentation commit which I think is in the correct place, describing the change here.

File: docs/source/release/v5.1.0/framework.rst [hunk 1]
```diff
@@ -32,6 +32,10 @@ The attenuation profile filename can also be specified in the materials section
 
 The sample environment xml file now supports the geometry being supplied in the form of a .3mf format file (so far on the Windows platform only). Previously it only supported .stl files. The .3mf format is a 3D printing format that allows multiple mesh objects to be stored in a single file that can be generated from many popular CAD applications. As part of this change the algorithms :ref:`LoadSampleEnvironment <algm-LoadSampleEnvironment>` and :ref:`SaveSampleEnvironmentAndShape <algm-SaveSampleEnvironmentAndShape>` have been updated to also support the .3mf format
 
+The :ref:`LoadISISNexus <algm-LoadISISNexus>` algorithm has been modified to remove the VMS compatibility block.
```
Comment (NickDraper) [inline docs/source/release/v5.1.0/framework.rst:35]:
```suggestion
The :ref:`LoadISISNexus <algm-LoadISISNexus>` algorithm has been modified to remove the need for the VMS compatibility block.
```

File: docs/source/release/v5.1.0/framework.rst [hunk 2]
```diff
@@ -32,6 +32,10 @@ The attenuation profile filename can also be specified in the materials section
...
+This has lead to the removal of the following variables from the sample logs as they were deemed unnecessary: dmp,
+dmp_freq, dmp_units dur, dur_freq, dur_secs, dur_wanted, durunits, mon_sum1, mon_sum2, mon_sum3, run_header.
```
Comment (NickDraper) [inline docs/source/release/v5.1.0/framework.rst:37]:
```suggestion
dmp_freq, dmp_units dur, dur_freq, dur_secs, dur_wanted, durunits, mon_sum1, mon_sum2, mon_sum3, run_header (this is available in the workspace title).
```

**Review by NickDraper — CHANGES_REQUESTED** [review]
A couple of minor changes. Then we are good to go.

**Review by NickDraper — APPROVED** [review]
All good to go, well done

**Review by NickDraper — CHANGES_REQUESTED** [review]
It looks like one of my sample logs tests has been merged into master that expects a certain number of sample logs, and the change here has caused the failure, you just need to adjust the number of logs expected.
```
======================================================================
FAIL: test_model_with_filtered_data (test_samplelogs_model.SampleLogsModelTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Jenkins/workspace/pull_requests-windows/qt/python/mantidqt/widgets/samplelogs/test/test_samplelogs_model.py", line 69, in test_model_with_filtered_data
    self.assertEqual(len(log_names), 152)
AssertionError: 140 != 152
```

**Comment by StephenSmith25** [comment]
> 
> 
> It looks like one of my sample logs tests has been merged into master that expects a certain number of sample logs, and the change here has caused the failure, you just need to adjust the number of logs expected.
> 
> ```
> ======================================================================
> FAIL: test_model_with_filtered_data (test_samplelogs_model.SampleLogsModelTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "C:/Jenkins/workspace/pull_requests-windows/qt/python/mantidqt/widgets/samplelogs/test/test_samplelogs_model.py", line 69, in test_model_with_filtered_data
>     self.assertEqual(len(log_names), 152)
> AssertionError: 140 != 152
> ```
I had to push two commits. The first fixed the failing tests, which was due to both the number of sample logs, but there was also a test failure due to trying to read "dur" out the logs. For that test I've replaced the call to "dur", to something else in the sample log hopefully that doesn't alter what the test was trying the "test". The second commit just removes an unused variable.

**Comment by NickDraper** [comment]
Ah yes 'dur' was not important, just a log picked at random

**Review by NickDraper — APPROVED** [review]
Good to go, well done

**Pull request merged**